bookkeeper-server/src/test/java/org/apache/bookkeeper/client/MockReadHandle.java

```diff
@@ -0,0 +1,141 @@
+    private final List<LedgerEntryImpl> entries;
+
+    MockReadHandle(MockBookKeeper bk, long ledgerId, LedgerMetadata metadata, List<LedgerEntryImpl> entries) {
+        this.bk = bk;
+        this.ledgerId = ledgerId;
+        this.metadata = metadata;
+        this.entries = entries;
+    }
+
+    @Override
+    public CompletableFuture<LedgerEntries> readAsync(long firstEntry, long lastEntry) {
+        CompletableFuture<LedgerEntries> promise = new CompletableFuture<>();
+        if (bk.isStopped()) {
+            promise.completeExceptionally(BKException.create(-1));
+            return promise;
+        }
+
+        bk.executor.execute(() -> {
+                if (bk.getProgrammedFailStatus()) {
+                    promise.completeExceptionally(BKException.create(bk.failReturnCode));
+                    return;
+                } else if (bk.isStopped()) {
+                    promise.completeExceptionally(BKException.create(-1));
+                    return;
+                }
+
+                log.debug("readEntries: first={} last={} total={}", firstEntry, lastEntry, entries.size());
+                List<LedgerEntry> seq = new ArrayList<>();
+                long entryId = firstEntry;
+                while (entryId <= lastEntry && entryId < entries.size()) {
+                    seq.add(entries.get((int) entryId++).duplicate());
+                }
+                log.debug("Entries read: {}", seq);
+                promise.complete(LedgerEntriesImpl.create(seq));
+            });
+        return promise;
+
+    }
+
+    @Override
+    public CompletableFuture<LedgerEntries> readUnconfirmedAsync(long firstEntry, long lastEntry) {
+        return readAsync(firstEntry, lastEntry);
+    }
+
+    @Override
+    public CompletableFuture<Long> readLastAddConfirmedAsync() {
+        return CompletableFuture.completedFuture((long) (entries.size() - 1));
+    }
+
+    @Override
+    public CompletableFuture<Long> tryReadLastAddConfirmedAsync() {
+        return readLastAddConfirmedAsync();
+    }
+
+    @Override
+    public long getLastAddConfirmed() {
+        return entries.size() - 1;
+    }
+
+    @Override
+    public long getLength() {
+        long length = 0;
+        for (LedgerEntryImpl entry : entries) {
+            length += entry.getLength();
+        }
+
+        return length;
+    }
+
+    @Override
+    public boolean isClosed() {
+        return metadata.isClosed();
+    }
+
+    @Override
+    public CompletableFuture<LastConfirmedAndEntry> readLastAddConfirmedAndEntryAsync(long entryId,
+                                                                                      long timeOutInMillis,
+                                                                                      boolean parallel) {
+        CompletableFuture<LastConfirmedAndEntry> promise = new CompletableFuture<>();
+        // it doesn't make sense to implement long poll until WriteHandle is also implemented
```

Comment:
the comment is misleading. long poll doesn't require write handle to implement. it is irrelative whether you use old write api or new write handle. please improve the comment or remove it.